templates/profile/vim/vimrc.vim.erb

```diff
@@ -12,43 +12,68 @@
 
-" Uncomment the next line to make Vim more Vi-compatible
-" NOTE: debian.vim sets 'nocompatible'.  Setting 'compatible' changes numerous
-" options, so any other options should be set AFTER setting 'compatible'.
-"set compatible
+" Remove ancient vi compatibility
+set nocompatible
 
-" Vim5 and later versions support syntax highlighting. Uncommenting the next
-" line enables syntax highlighting by default.
-syntax on
-
-" If using a dark background within the editing area and syntax highlighting
-" turn on this option as well
+" Color
+syntax enable
 set background=dark
 
-" Uncomment the following to have Vim jump to the last position when
-" reopening a file
-"if has("autocmd")
-"  au BufReadPost * if line("'\"") > 1 && line("'\"") <= line("$") | exe "normal! g'\"" | endif
-"endif
+" Load rules based on filetype
+filetype plugin indent on
 
-" Uncomment the following to have Vim load indentation rules and plugins
-" according to the detected filetype.
-if has("autocmd")
-  filetype plugin indent on
-endif
+" Search
+" disable vim-specific search regex by inserting \v
+nnoremap / /\v
+vnoremap / /\v
+set ignorecase	" Do case insensitive matching
+set smartcase		" Unless we mix cases
+set gdefault    " global search by default
+set incsearch		" Incremental search
+set hls         " highlight search as we type
+set showmatch   " Show all matches at once
 
-" The following are commented out as they cause vim to behave a lot
-" differently from regular Vi. They are highly recommended though.
-set showcmd		" Show (partial) command in status line.
+" Improve usability
+set showcmd		  " Show (partial) command in status line.
 set showmatch		" Show matching brackets.
-set ignorecase		" Do case insensitive matching
-set smartcase		" Do smart case matching
-set incsearch		" Incremental search
 set autowrite		" Automatically save before commands like :next and :make
-set hidden             " Hide buffers when they are abandoned
+set hidden      " Hide buffers when they are abandoned
 " set mouse=a		" Enable mouse usage (all modes)
 set mouse=
 
+" Disable sounds
+set noerrorbells                " shhh
+set novisualbell                " shhh
+set tm=500
+
+" Smart indent when entering insert mode on empty lines
+function! IndentWithI()
+  if len(getline('.')) == 0
+    return "\"_cc"
+  else
+    return "i"
+  endif
+endfunction
+nnoremap <expr> i IndentWithI()
+
+" Highlight and remove trailing whitespace
+autocmd BufEnter * EnableStripWhitespaceOnSave
+highlight ExtraWhitespace ctermbg=darkgreen guibg=darkgreen
+
+" tabs
+set expandtab                   " use spaces
+set smarttab                    " be smort
+set shiftwidth=2                " 2 spaces
+set tabstop=2                   " 2 spaces!
+set softtabstop=2               " 2 spaces!!!!
+set autoindent                  " enable automatic indentation
+set smartindent                 " does the right thing
+
+" Disable backups that just clutter git
+set nobackup
+set nowb
+set noswapfile
```

Comment:
these options might be controversial... I would check in #dev before setting these.

Reply:
I totally forgot that this applied to devs as well.